classes/userGroup/Repository.php

```diff
@@ -147,6 +149,11 @@ public function add(UserGroup $userGroup): int
 
         Hook::call('UserGroup::add', [$userGroup]);
 
+        // Clear editorial masthead cache if the new role should be added to the masthead
+        // Because it is a new role, no need to clear editorial history chache
+        if ($userGroup->getMasthead()) {
+            AboutContextHandler::forgetEditorialMastheadCache($userGroup->getContextId());
```

Comment:
Rather than having the user group repository (business logic) reach out to the presentation layer to clear the cache, how about moving the caching code directly here, and having the AboutContextHandler call it from here instead? That would keep the model layer a little cleaner and less tangled up with presentation.

Reply:
Yes, that makes perfectly sense! Thanks!